server/server.go

```diff
@@ -318,6 +345,17 @@ func (s *Server) createTLSConfig(entryPointName string, tlsOption *traefiktls.TL
 		return nil, err
 	}
 
+	if tlsOption.DefaultCertificate != nil {
```

Comment:
Before this code, the method `certificate.CreateTLSConfig` is called [here](https://github.com/containous/traefik/blob/master/server/server.go#L316).

It generates a `by default` self signed certificate if necessayr and add all the static certificates (or the generated one) to the `tls.Config.Certificates` list.
IMHO, the check on the default certificate can be done before to generate a sefl-signed certificate and,in addition, the `tls.Config.Certificates` list has not to be set because you code allows Træfik to manage entirely the `getCertificate` mechanism and the `tls.Config.Certificates` will not be used.

WDYT?

Reply:
The problem is that `tlsOption.Certificates.CreateTLSConfig(entryPointName)` is called on the `tls.Config.Certificates` slice. It does not allow for `tls.Config.DefaultCertificate` to be added in if present.

Yes. You are correct. This was from before I handled static certs. This is now irrelevant, and will be removed.

Reply:
Re-instating this, because I have found out that `getCertificate` does not always get called ;)